File: dotnet/DesignPatterns/BehavioralDesignPatterns/MediatorDesignPattern/MediatorDesignPattern.Example2/Handlers/GetAllOrdersQueryHandler.cs
using MediatR;
using MediatorDesignPattern.Example2.Models;
using MediatorDesignPattern.Example2.Queries;
using MediatorDesignPattern.Example2.Repositories;

namespace MediatorDesignPattern.Example2.Handlers;

/// <summary>
/// Query handler that retrieves all orders
/// </summary>
public class GetAllOrdersQueryHandler : IRequestHandler<GetAllOrdersQuery, List<Order>>
{
    private readonly IOrderRepository _orderRepository;

    public GetAllOrdersQueryHandler(IOrderRepository orderRepository)
    {
        _orderRepository = orderRepository;
    }

    public Task<List<Order>> Handle(GetAllOrdersQuery request, CancellationToken cancellationToken)
    {
        var orders = _orderRepository.GetAll();
        
        Console.WriteLine($"[HANDLER] {orders.Count} orders retrieved");
        
        return Task.FromResult(orders);
    }
}

File: nodejs/DesignPatterns/BehavioralDesignPatterns/MediatorDesignPattern/MediatorDesignPattern.Example2/src/Handlers/GetAllOrdersQueryHandler.ts
import { requestHandler, RequestHandler } from 'mediatr-ts';
import { GetAllOrdersQuery } from '../Queries/GetAllOrdersQuery';
import { Order } from '../Models/Order';
import { orderRepository } from './CreateOrderCommandHandler';

@requestHandler(GetAllOrdersQuery)
export class GetAllOrdersQueryHandler implements RequestHandler<GetAllOrdersQuery, Order[]> {
  async handle(_request: GetAllOrdersQuery): Promise<Order[]> {
    const orders = orderRepository.getAll();
    console.log(`[HANDLER] ${orders.length} orders retrieved`);
    return orders;
  }
}
